### PR TITLE
[Feature] Add smpl visualization and mask in smpl data

### PR DESCRIPTION
**Pull request by LazyBusyYang**
1. Fix lint error caused by isort.
2. Clarify difference between `param_dict` and `dict(smpl_data)` by removing `src_dict` and add `from_dict()`.
3. Add mask in smpl_data. The person whose mask is zero will not be plotted.
5. Fix wrong gender type when calling `SMPLData.fromfile()`
4. Add smpl visualization and unit test, based on `minimal_pytorch_rasterizer`. Multi-person and multi-gender are supported.
5. Update dockerfile and docker image, as well as install.md.
6. Some designed features have not been enabled. They will not be OK until their requirements in XRPrimer is updated.

### CODE REVIEW

File: xrmocap/data_structure/body_model/smpl_data.py [hunk 1]
```diff
@@ -60,14 +55,16 @@ def __init__(self,
                 in shape [n_frame, betas_dim].
                 Defaults to None,
                 zero-tensor in shape [n_frame, 10] will be created.
+            mask (Union[np.ndarray, torch.Tensor, None], optional):
```
Comment (yl-1993) [inline xrmocap/data_structure/body_model/smpl_data.py:58]: `mask` is frame-level mask?

File: docs/en/installation.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ Optional:
 | [MMTracking](https://github.com/open-mmlab/mmtracking)   | Multiple object tracking.      | Install `mmcv-full`, instead of `mmcv`.                      |
 | [MMDeploy](https://github.com/open-mmlab/mmdeploy)       | Faster mmdet+mmpose inference. | Install `mmcv-full`, `cudnn` and `TensorRT`.                 |
 | [Aniposelib](https://github.com/google/aistplusplus_api) | Triangulation.                 | Install from [github](https://github.com/liruilong940607/aniposelib), instead of pypi. |
+| [Minimal Pytorch Rasterizer](https://github.com/rmbashirov/minimal_pytorch_rasterizer) | SMPL mesh fast visualization.                 | Sensitive to pytorch version. |
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply docs/en/installation.md:31]: In [official repo](https://github.com/rmbashirov/minimal_pytorch_rasterizer) the author says 'Tested till pytorch 1.8'. We've tested it on pytorch1.12.0(same as a-from-scratch-setup-script) and released the good docker image. No more tests have been done.

File: xrmocap/core/visualization/visualize_smpl.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+    import_exception = stack_str + import_exception
+# yapf: enable
+
+
+def visualize_smpl_data(
+        # input args
+        smpl_data: Union[SMPLData, SMPLXData, List[Union[SMPLData,
+                                                         SMPLXData]]],
+        body_model: Union[SMPL, SMPLX, dict, List[Union[SMPL, SMPLX, dict]]],
+        cam_param: Union[FisheyeCameraParameter, PinholeCameraParameter],
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply xrmocap/core/visualization/visualize_smpl.py:42]: This function visualize smpl from a fix view. In this case, there will be an auto camera. Coming in the following commits.

File: xrmocap/data_structure/body_model/smpl_data.py (hunk 1 above)
Comment (LazyBusyYang) [reply xrmocap/data_structure/body_model/smpl_data.py:58]: Yes. Each smpl data holds data for only one person, and we do not mask parts or joints.

File: xrmocap/core/visualization/visualize_smpl.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+                         'the latest xrprimer release is ' + 'not correct.')
+            raise NotImplementedError
+        else:
+            array_to_video(image_array=img_arr, output_path=output_path)
+        if remove_output_dir:
+            shutil.rmtree(output_dir)
+    return img_arr if return_array else None
+
+
+class MPRNormRenderer:
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply xrmocap/core/visualization/visualize_smpl.py]: Fixed in [091d69e](https://github.com/openxrlab/xrmocap/pull/73/commits/091d69e11cf8ab22512d7da2d0bdb54808806894).

File: tests/core/visualization/test_visualize_smpl.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    # array_to_video(image_array=black_background, output_path=bg_video_path)
+    # output_video = os.path.join(output_dir, 'background_video.mp4')
+    # visualize_smpl_data(
+    #     smpl_data=smpl_data_list,
+    #     body_model=neutral_model,
+    #     cam_param=pinhole_param,
+    #     output_path=output_video,
+    #     overwrite=True,
+    #     background_video=bg_video_path,
+    # )
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply tests/core/visualization/test_visualize_smpl.py]: It's no debug info. Add condition in [091d69e](https://github.com/openxrlab/xrmocap/pull/73/commits/091d69e11cf8ab22512d7da2d0bdb54808806894).

File: xrmocap/data_structure/body_model/smplx_data.py [hunk 6]
```diff
@@ -68,24 +63,57 @@ def __init__(self,
...
+        Returns:
+            SMPLXData:
+                A SMPLXData instance load from dict.
+        """
+        smplx_data_dict = smpl_data_dict
+        assert 'gender' in smplx_data_dict
+        assert 'fullpose' in smplx_data_dict
+        assert 'transl' in smplx_data_dict
+        assert 'betas' in smplx_data_dict
+        assert 'expression' in smplx_data_dict
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply xrmocap/data_structure/body_model/smplx_data.py]: Fixed in [718f4c1](https://github.com/openxrlab/xrmocap/pull/73/commits/718f4c138eb69d9cad498d08f94c330639c429ad).

File: xrmocap/core/visualization/visualize_smpl.py [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+        default_body_model = model
+    elif isinstance(body_model, list):
+        body_model_list = body_model
+        for body_model in body_model_list:
+            if isinstance(body_model, dict):
+                model = build_body_model(body_model).to(device)
+                body_model_dict[model.gender] = model
+            else:
+                model = body_model.to(device)
+                body_model_dict[model.gender] = model
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply xrmocap/core/visualization/visualize_smpl.py]: Fixed in [794c3cb](https://github.com/openxrlab/xrmocap/pull/73/commits/794c3cb10fa5f5e5e831289bc012425e89db80c9).

File: xrmocap/data_structure/body_model/smplxd_data.py [hunk 8]
```diff
@@ -62,25 +57,60 @@ def __init__(self,
...
+            SMPLXData:
+                A SMPLXDData instance load from dict.
+        """
+        smplxd_data_dict = smpl_data_dict
+        assert 'gender' in smplxd_data_dict
+        assert 'fullpose' in smplxd_data_dict
+        assert 'transl' in smplxd_data_dict
+        assert 'betas' in smplxd_data_dict
+        assert 'expression' in smplxd_data_dict
+        assert 'displacement' in smplxd_data_dict
```
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply xrmocap/data_structure/body_model/smplxd_data.py]: Fixed in [718f4c1](https://github.com/openxrlab/xrmocap/pull/73/commits/718f4c138eb69d9cad498d08f94c330639c429ad).

File: docs/en/installation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yl-1993) [reply docs/en/installation.md:31]: Great. Why not recording all tested versions? like `Sensitive to PyTorch version. PyTorch<= 1.8 and ==1.12 have been tested.`

File: xrmocap/data_structure/body_model/smpl_data.py (hunk 1 above)
Comment (yl-1993) [reply xrmocap/data_structure/body_model/smpl_data.py:58]: Shall we doc it as `A tensor or ndarray for frame-level visibility mask` or rename it to `frame_mask`?

File: docs/en/installation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply docs/en/installation.md:31]: I guess that 1.9, 1.10 and 1.10 also work. Typically there's no significant differences between a newer and an older version, but we still need time to prove it.   
Also, the same docker image works on 1660ti, 3080 but fails on 1080ti. I do not have an explicit conclusion for MPR compatibility.

File: xrmocap/data_structure/body_model/smpl_data.py (hunk 1 above)
Comment (LazyBusyYang) [reply xrmocap/data_structure/body_model/smpl_data.py:58]: We won't have another mask for joints or parts, will we? Considering the parents and children of joints, it is not possible to mask an independent joint.
By setting a keypoint's visibility to 0, we directly remove it from visualization. But for a joint, we can only set it to zero and make it look like an initiative pose, visibility has not been changed.

Comment (LazyBusyYang) [reply xrmocap/data_structure/body_model/smpl_data.py:58]: Docstring fixed in [fb637eb](https://github.com/openxrlab/xrmocap/pull/73/commits/fb637ebd1bc617ef874ee4f15dc0b6dd57ba3c8f).

File: docs/en/installation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (LazyBusyYang) [reply docs/en/installation.md]: Docs fixed in [fb637eb](https://github.com/openxrlab/xrmocap/pull/73/commits/fb637ebd1bc617ef874ee4f15dc0b6dd57ba3c8f).

**Review by yl-1993 — APPROVED** [review]

**Pull request merged**